src/os_win.cc

```diff
@@ -56,10 +55,14 @@ static char* malloced_narrow_string(const wchar_t* string) {
 }
 
 char* OS::get_executable_path() {
-  char* path = _new char[MAX_PATH + 1];
-  auto length = GetModuleFileName(NULL, path, MAX_PATH);
-  path[length] = '\0';
-  return path;
+  const int BUFFER_SIZE = 32767 + 1;
```

Comment:
This feels like too much to put on the stack. I'd just use malloc and expect it to return a non-null value.

Reply:
I copied this from another function.
Will do a follow-up PR where I change both to mallocs.